### PR TITLE
Debug Build Support

### LINKED ISSUE #604: Allow the creation of an MCprep build with proper debugging
As MCprep is an extremely complex addon, debugging using traditional means in Blender is incredibly difficult. Some of the bugs in #601 were extremely deep level bugs that were incredibly difficult to debug normally. Each time we wanted to check a different variable, we had to modify print statements and rebuild the code, which is slow when a bug involves assets.

In an ideal world, Blender would come with a Python debugger that would allow us to analyze multiple things at once, and would allow the use of breakpoints. With the addition of Blender Extensions however, it might be possible to allow the creation of a debug build that uses https://github.com/microsoft/debugpy, and allow us to attach to it from an editor that supports it (VScode natively, Neovim with https://github.com/HiPhish/debugpy.nvim, etc). Then we could add a `debug-build` action that includes these wheels into the MCprep build, and otherwise excludes them at runtime. This would allow for much more advanced debugging in MCprep.

Some requirements:
- Any version of Blender with Python 3.8+ (as of writing); Blender 4.2 should be fine with this requirement
- The ability to build MCprep as an extension; this is possible right now in BpyBuild, so long as we include a `blender_manifest.toml` file in the `MCprep_addon` directory (though manifest verification has [not yet been merged](https://github.com/Moo-Ack-Productions/bpy-build/pull/19) in BpyBuild)
	- The reason the ability to use MCprep as an extension is important is because Blender now natively supports using third-party packages in extensions by [bundling wheels](https://docs.blender.org/manual/en/4.2/extensions/python_wheels.html) into the extension.

### PR DESCRIPTION
**Pull request by StandingPadAnimations**
This adds the ability to create a build of MCprep that can then be debugged using a debugger. Under the hood, this uses Microsoft's [debugpy](https://github.com/microsoft/debugpy) library, which is then installed into the Blender environment using the new wheels system introduced with Extensions (support for which has also been added).

This has been tested with VScode, but in theory should be editor agnostic. Some notes to be aware of:
- MCprep must be manually installed as a debug build, because installing extensions isn't supported in BpyBuild yet
- When using a debugger, the editor must be opened in the directory of the installed extension (ex. `~/.config/blender/4.2/extensions/user_default/mcprep/`). Hopefully we can get this ironed out by either including some configs or having the `debug` action generate them on the fly.
- To build with debug support, run `bab -b dev extension debug`
- Additional configuration might be required. For example, with VScode, the following configuration needs to be used:
```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Python Debugger: Remote Attach",
            "type": "debugpy",
            "request": "attach",
            "connect": {
                "host": "localhost",
                "port": 5678
            },
            "pathMappings": [
                {
                    "localRoot": "${workspaceFolder}",
                    "remoteRoot": "${workspaceFolder}"
                }
            ]
        }
    ]
}
```
- This is not yet documented as it's a pretty advanced development feature, there really isn't an easy way to document it (especially as something editor agnostic)

Closes #604

### CODE REVIEW

**Comment by TheDuckCow** [comment]
Just glancing through, I see this also represents embracing the new extension format. I gather that would mean needing to distribute two versions of MCprep right, one in extension format and one as a legacy addon?

**Comment by StandingPadAnimations** [comment]
We could if we want to, or we could keep using only legacy builds for production. For simplicity, we should just distribute legacy builds, but make sure extension builds keep working for debug build support or for users that want to build an extension instead.

**Review by TheDuckCow — CHANGES_REQUESTED** [review]
Main reason for requesting change is to see if I can get the extension build to properly install the debuggy wheel. Same thing happens in 4.2 and 4.3, whether dragging and dropping to install the extension, navigating manually; and same popup appears when trying to manually enable the extension's addon after the extension is installed.

Other points are less important, figure we'd want to make sure this works cross platform, even though I personally probably won't end up using it much.

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+    // Use IntelliSense to learn about possible attributes.
```
[the remark this answers is not in the document]
Comment (StandingPadAnimations) [reply .vscode/launch.json:2]: This was automatically generated by VScode, I think VScode config files are just weird with supported JSON features

File: action-scripts/build-patches/debug-mode.diff [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+@@ -17,5 +17,11 @@ license = [
+   "SPDX:GPL-3.0-or-later",
+ ]
+ 
++wheels = [
++  "./wheels/debugpy-1.8.2-cp311-cp311-macosx_11_0_universal2.whl",
++  "./wheels/debugpy-1.8.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl",
++  "./wheels/debugpy-1.8.2-cp311-cp311-win_amd64.whl",
++]
++
```
[the remark this answers is not in the document]
Comment (StandingPadAnimations) [reply action-scripts/build-patches/debug-mode.diff:14]: Strange, the macOS wheel should be universal across both Intel and Silicon. I'll take a look at the action

**Comment by TheDuckCow** [comment]
(Also, while not related to the v3.6.1 release, could be nice to merge by then I'd say since this seems pretty close otherwise)

File: action-scripts/debug-build.py
[the remark this answers is not in the document]
Comment (StandingPadAnimations) [reply action-scripts/debug-build.py:1]: There are plans for BpyBuild to handle extensions properly (already implemented technically, but some things still need to be finished, such as building legacy addons and extensions at the same time). Compatibility minimums would be something interesting to add for 0.5

**Review by TheDuckCow — APPROVED** [review]
Approved based on the wheel change we did for osx, knowing we might go a bit further in the future (I won't be using this mode in the meantime anyways)

**Pull request merged**